src/ai-http-request/infrastructure/api-client.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Yoast\WP\SEO\AI_HTTP_Request\Infrastructure;
+
+use WPSEO_Utils;
+use Yoast\WP\SEO\AI_HTTP_Request\Domain\Exceptions\WP_Request_Exception;
+
+/**
+ * Class API_Client
+ * Handles the API requests to the AI Generator API.
+ */
+class API_Client implements API_Client_Interface {
+
+	/**
+	 * The base URL for the API.
+	 *
+	 * @var string
+	 */
+	private $base_url;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param string $base_url The base URL for the API.
+	 */
+	public function __construct( string $base_url ) {
+		$this->base_url = $base_url;
+	}
+
+	/**
+	 * Performs a request to the API.
+	 *
+	 * @param string        $action_path The action path for the request.
+	 * @param array<string> $body        The body of the request.
+	 * @param array<string> $headers     The headers for the request.
+	 * @param bool          $is_post     Whether the request is a POST request.
+	 *
+	 * @return array<int|string|array<string>> The response from the API.
+	 *
+	 * @throws WP_Request_Exception When the wp_remote_post() returns an error.
+	 */
+	public function perform_request( string $action_path, $body, $headers, bool $is_post ): array {
```

Comment:
I don't like the `is_post` here. using if's to change behaviour is always a bit weird. `perform_get` or just `get` and `post` would not be bad here I think.